docs/proposals/27-falco-extension.md

```diff
@@ -0,0 +1,352 @@
+changing host configurations or installing software in containers. Apart from
+legal debugging activities those types of events most likely indicate malicious
+behavior.
+
+Falco is a powerful runtime threat detection engine but comes with limited
+means to deploy on many Kubernetes clusters or sophisticated event analysis
+and visualization for many clusters. These features are usually provided by
+commercial products: some of them rely on Falco as the threat detection
+engine, for example from [Sysdig](https://sysdig.com/products/platform/) and
+[Trend Micro](https://www.trendmicro.com/en_us/business/products/one-platform.html).
+
+This enhancement proposal describes a consistent way for deploying Falco into
+the Gardener infrastructure as well as into shoot clusters operated by Gardener
+users in a "Gardener native" way. This includes managing it, providing Falco
+updates and bringing a central infrastructure for viewing and analyzing events.
+
+## Motivation
+
+While we have invested into Gardener security and carry out regular
+penetration tests we currently do not have sufficient runtime detection of 
+potentially malicious activities in our Kubernetes clusters, unless we install
+3rd party commercial tools which some of our users do. Malicious activities
+include but are not limited to security breaches through misconfigurations or
+0-day vulnerabilities.
+
+We are planning to close this gap by deploying the Falco runtime security
+tool.
+
+### Goals
+
+- Deploy Falco into the Gardener infrastructure.
+- Allow Gardener users to deploy Falco into their shoot clusters by adding a
+  configuration into their shoot manifests.
+- View and analyze Falco events in the Gardener dashboard.
+- Deploy an optimized ruleset for Gardener managed Kubernetes clusters based 
+  on the Falco rules [5]. Optionally allow deployment of incubating-, and
+  sandbox rules as well as custom rules.
+- Offer multiple Falco versions to users including options for manual- and 
+  automated updates.
+- Allow users to simply turn on Falco in their clusters with a reasonable 
+  default configuration as well as allow power users to fine tune the 
+  configuration (rules, plug-ins).
+- Investigate existing Falco plug-ins or develop new ones in the Gardener
+  context to broaden the detection scope
+- Provide means for users to configure Falco themselves using `falcoctl` and
+  configure their own event store.
+
+### Non-Goals
+
+We do not compete with commercial products. While we have plans to develop
+Falco plugins, our goal is not to provide a comprehensive solution provided
+by security vendors.
+
+## Proposal
+
+We are planning to provide a Gardener extension that deploys Falco and 
+Falcosidekick into shoot clusters. Falco is deployed as a Kubernetes DaemonSet
+on each worker node and Falcosidekick is deployed as a Kubernetes Deployment. 
+All events that are generated by Falco are sent to a central event store 
```

Comment:
Will it be possible to split this central event store? I am asking because it is possible with single Gardener to run shoots in different networks, hence some of them might not have the network visibility to the event store.